be/src/vec/common/pod_array.h

```diff
@@ -120,8 +122,9 @@ class PODArrayBase : private boost::noncopyable,
         }
     }
 
+    /// Not round up, keep the size just as the application pass in like std::vector
     void alloc_for_num_elements(size_t num_elements) {
-        alloc(round_up_to_power_of_two_or_zero(minimum_memory_for_elements(num_elements)));
+        alloc(minimum_memory_for_elements(num_elements));
```

Comment:
Allocating in powers of 2 has a positive impact on performance, if you wish to reduce memory usage,

join `#ifndef STRICT_MEMORY_USE`, similar to `hash_table.h` expansion

Reply:
I do not think this should be a config, because if it is a config, we do not know when to open the config since it is a macro. Actually, there are two types of memory allocation in PODArray:
1. reserve, sometimes the developer know the expected size of the array, then he should call reserve method to allocate the EXPECTED memory.
2. push_back, the developer does not know the expected size of the array, then he just call push back to allocate memory. In this scenario, we should allocate memory using power of 2.

For most cases, we should reserve or resize memory size before push back, then we could reduce memory reallocation or memory copy. 
This PR try to fix some problems. https://github.com/apache/doris/pull/13088.